website/docs/r/edgegateway.html.markdown

```diff
@@ -0,0 +1,50 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_edgegateway"
+sidebar_current: "docs-vcd-resource-edgegateway"
+description: |-
+  Provides a vCloud Director edge gateway. This can be used to create and delete edge gateways connected to one or more external networks.
+---
+
+# vcd\_edgegateway
+
+Provides a vCloud Director edge gateway directly connected to one or more external networks. This can be used to create
+and delete edge gateways for Org VDC networks to connect.
+
+Supported in provider *v2.4+*
+
+~> **Note:** Only `System Administrator` can create an edge gateway.
+You must use `System Adminstrator` account in `provider` configuration
+and then provide `org` and `vdc` arguments for edge gateway to work.
+
+## Example Usage
+
+```hcl
+resource "vcd_edgegateway" "egw" {
+  org = "my-org"
+  vdc = "my-vdc"
+
+  name              = "my-egw"
+  description       = "new edge gateway"
+  backing_config    = "compact"
+  default_gateway   = "my-ext-net1"
+  external_networks = [ "my-ext-net1", "my-ext-net2" ]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Required) The name of organization to which the VDC belongs.
+* `vdc` - (Required) The name of VDC that owns the edge gateway.
+* `name` - (Required) A unique name for the edge gateway.
+* `external_networks` - (Required) An array of external network names.
+* `backing_config` - (Required) Configuration of the vShield edge VM for this gateway. One of: `compact`, `full`.
```

Comment:
I don't know but maybe we should add note that we support subset of configs available in UI. cc:  @lvirbalas

Reply:
We should add them when we are sure that we can support them.
Our current API doesn't seem to support more values. When this change, we'll update the documentation. Nothing more needs to happen, except perhaps a few more tests

Reply:
"Our current API" == current API version we are handshaking with?

Reply:
yes

Reply:
Added the full list of parameters